src/main/java/view/createPostView.java

```diff
@@ -0,0 +1,489 @@
+            }
+
+            public void mouseExited(java.awt.event.MouseEvent evt) {
+                button.setBackground(color);
+            }
+        });
+        return button;
+    }
+
+    private void addPostTitleListener() {
+        titleField.getDocument().addDocumentListener(new DocumentListener() {
+            private void documentListenerHelper() {
+                final CreatePostState currentState = createPostViewModel.getState();
+                currentState.setPostTitle(titleField.getText());
+                createPostViewModel.setState(currentState);
+            }
+
+            public void insertUpdate(DocumentEvent e) {
+                documentListenerHelper();
+            }
+
+            public void removeUpdate(DocumentEvent e) {
+                documentListenerHelper();
+            }
+
+            public void changedUpdate(DocumentEvent e) {
+                documentListenerHelper();
+            }
+        });
+    }
+
+    private void addPostContentListener() {
+        contentField.getDocument().addDocumentListener(new DocumentListener() {
+            private void documentListenerHelper() {
+                final CreatePostState currentState = createPostViewModel.getState();
+                currentState.setContent(contentField.getText());
+                createPostViewModel.setState(currentState);
+            }
+
+            public void insertUpdate(DocumentEvent e) {
+                documentListenerHelper();
+            }
+
+            public void removeUpdate(DocumentEvent e) {
+                documentListenerHelper();
+            }
+
+            public void changedUpdate(DocumentEvent e) {
+                documentListenerHelper();
+            }
+        });
+    }
+
+    private void addSaveButtonListener() {
+        saveButton.addActionListener(e -> {
+            String author = homepageController.fetchUser().getUsername();
+            String content = contentField.getText();
+            String attachmentPath = "";
+            String fileType = "text";
+            int dislikes = 0;
+            int likes = 0;
+            String postTitle = titleField.getText();
+            java.util.List<User> moderators = new ArrayList<>();
+            java.util.List<Comment> comments = new ArrayList<>();
+            String category = (String) categoryDropdown.getSelectedItem();
+
+            if (postTitle.isEmpty() || content.isEmpty() || author.isEmpty()) {
+                JOptionPane.showMessageDialog(this, "Title, content, and author are required!");
+                return;
+            }
+
+            createPostController.execute(homepageController.fetchUser().getUserID(), content, attachmentPath, fileType, dislikes, likes, postTitle, moderators, comments, category);
+            JOptionPane.showMessageDialog(this, "Post created successfully!");
+            SwingUtilities.getWindowAncestor(this).dispose();
+            createPostController.switchToHomePageview();
```

Comment:
Think you need to add homepageController.fetchAllPosts(); before switchtohomepageview?

Reply:
thanks! just added that in